spec.bs

```diff
@@ -201,68 +208,250 @@ This document introduces a new storage API that is intentionally not partitioned
+            1. Let |operationMap| be the associated {{SharedStorageWorkletGlobalScope}}'s [=SharedStorageWorkletGlobalScope/operation map=].
+            1. If |operationMap| [=map/contains=] |operationName|:
+                1. Let |operation| be |operationMap|[|operationName|].
+                1. Let |argumentsList| be the [=/list=] « |urlList| ».
+                1. If |options| [=map/contains=] |data|, [=list/append=] |data| to |argumentsList|.
+                1. Let |operationResult| be the result of [=invoking=] |operation| with |argumentsList|.
+                1. If [=an exception was thrown=], then [=queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}.
+                1. Otherwise:
+                    1. Set |index| to |operationResult|, [=converted to a JavaScript value=] of type {{unsigned long}}.
+                    1. If [=an exception was thrown=]:
+                        1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}.
+                        1. Abort these steps.
+                    1. Otherwise, if |index| is greater than |urlList|'s [=list/size=]:
+                        1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=reject=] |promise| with a {{TypeError}}.
+                        1. Abort these steps.
+                    1. [=Queue a global task=] on the [=DOM manipulation task source=], given |window|, to [=resolve=] |promise| with |index|.
+  </div>
+
+  <div algorithm>
+    The <dfn method for="SharedStorageWorklet">selectURL(|name|, |urls|, |options|)</dfn> method steps are:
+
+    1. Let |resultPromise| be a new [=promise=].
+    1. If [=this=]'s [=addModule initiated=] is false, then return a [=promise rejected=] with a {{TypeError}}.
+    1. Let |window| be the [=current global object=].
+    1. [=Assert=]: |window| is a {{Window}}.
+    1. Let |context| be |window|'s [=Window/browsing context=].
+    1. If |context| is null, then return a [=promise rejected=] with a {{TypeError}}.
+    1. Let |document| be |context|'s [=active document=].
+    1. If the result of running [=Is feature enabled in document for origin?=] on "[=PermissionsPolicy/shared-storage-select-url=]", |document|, and [=this=]'s [=SharedStorageWorklet/worklet origin=] returns false, return a [=promise rejected=] with a {{TypeError}}.
+    1. If [=this=]'s [=global scopes|list of global scopes=] is [=list/empty=], then return a [=promise rejected=] with a {{TypeError}}.
+    1. [=Assert=]: [=this=]'s [=global scopes|list of global scopes=] [=list/contains=] a single {{SharedStorageWorkletGlobalScope}}.
+    1. If the result of running [=SharedStorageWorkletGlobalScope/check whether addModule is finished=] for [=this=]'s {{SharedStorageWorkletGlobalScope}} is false, return a [=promise rejected=]  with a {{TypeError}}.
```

Comment:
"this's SharedStorageWorkerGlobalScope" is not a real member. Do you mean to grab the global scope from the list of global scopes, associated with the outer worklet class (that hangs off of `Window`)?